### PR TITLE
Multi label graph interface

### PR DESCRIPTION
**Pull request by andyfengHKU**
# Description

This PR supports multi label graph in `Graph` interface.

# Contributor agreement

- [x] I have read and agree to the [Contributor Agreement](https://github.com/kuzudb/kuzu/blob/master/CLA.md).

### CODE REVIEW

File: src/graph/on_disk_graph.cpp [hunk 1]
```diff
@@ -10,52 +10,132 @@ using namespace kuzu::common;
...
+            }
+            fwdRelTables.insert({relTableID, storage->getTable(relTableID)->ptrCast<RelTable>()});
+        }
+        nodeTableIDToFwdRelTables.insert({nodeTableID, std::move(fwdRelTables)});
+        common::table_id_map_t<storage::RelTable*> bwdRelTables;
+        for (auto& relTableID : nodeTableEntry.getBwdRelTableIDSet()) {
+            if (!graphEntry.containsRelTableID(relTableID)) {
+                continue;
+            }
+            bwdRelTables.insert({relTableID, storage->getTable(relTableID)->ptrCast<RelTable>()});
```
Comment (semihsalihoglu-uw) [inline src/graph/on_disk_graph.cpp:59]: When we develop the on-disk-graph, do we construct a version with both forward and backward edges? Many graph algorithms only need one direction.

If we have a TODO list for algorithms package, can you add "Support for constructing graphs (on-disk and in-mem) that only contain one directional relationships."

File: src/planner/plan/plan_read.cpp [hunk 2]
```diff
@@ -146,6 +146,18 @@ void Planner::planGDSCall(const BoundReadingClause& readingClause,
             planReadOp(getGDSCall(readingClause), predicatesToPush, *plan);
         }
     }
+    if (bindData->hasNodeOutput()) {
```
Comment (semihsalihoglu-uw) [inline src/planner/plan/plan_read.cpp:149]: What's this change about? Is it connected to multi-label graphs or something independent?

File: src/processor/map/map_semi_masker.cpp [hunk 3]
```diff
@@ -44,9 +44,11 @@ std::unique_ptr<PhysicalOperator> PlanMapper::mapSemiMasker(LogicalOperator* log
             KU_ASSERT(physicalOp->getChild(0)->getOperatorType() == PhysicalOperatorType::GDS_CALL);
             auto gds = physicalOp->getChild(0)->ptrCast<GDSCall>();
             KU_ASSERT(gds->hasSemiMask());
-            auto mask = gds->getSemiMask();
-            auto tableID = mask->getNodeTable()->getTableID();
-            masksPerTable.at(tableID).emplace_back(mask, 0);
+            // TODO(Xiyang): we can refactor this logic with RECURSIVE JOIN above.
```
Comment (semihsalihoglu-uw) [inline src/processor/map/map_semi_masker.cpp:47]: Do you want to address this TODO while you're at it? You might leave algorithms anyways.

**Review by semihsalihoglu-uw — APPROVED** [review]

File: src/planner/plan/plan_read.cpp (hunk 2 above)
Comment (andyfengHKU) [reply src/planner/plan/plan_read.cpp:149]: I'll soon open a PR to make algorithm not only returns node_id but also node properties.

File: src/processor/map/map_semi_masker.cpp (hunk 3 above)
Comment (andyfengHKU) [reply src/processor/map/map_semi_masker.cpp:47]: I'll do this in a separate PR.

**Pull request merged**